packages/spacecat-shared-data-access/test/it/v2/index.test.js

```diff
@@ -0,0 +1,241 @@
+      status: 'IN_PROGRESS',
+    };
+
+    await opportunity
+      .setRunbook(updates.runbook)
+      .setStatus(updates.status)
+      .save();
+
+    // validate in-memory updates
+    expect(opportunity.getRunbook()).to.equal(updates.runbook);
+    expect(opportunity.getStatus()).to.equal(updates.status);
+
+    // validate unchanged fields
+    const {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      runbook, status, updatedAt, ...originalUnchangedFields
+    } = sampleData.opportunities[0].record;
+    const {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      runbook: _, status: __, updatedAt: ___, ...actualUnchangedFields
+    } = opportunity.record;
+
+    expect(actualUnchangedFields).to.eql(originalUnchangedFields);
+
+    // validate persistence of updates
+    const storedOpportunity = await Opportunity.findById(sampleData.opportunities[0].getId());
+    expect(storedOpportunity.getRunbook()).to.equal(updates.runbook);
+    expect(storedOpportunity.getStatus()).to.equal(updates.status);
+
+    // validate timestamps or audit logs
+    expect(new Date(storedOpportunity.record.updatedAt)).to.be.greaterThan(
+      new Date(sampleData.opportunities[0].record.updatedAt),
+    );
+
+    // validate persisted record matches in-memory state
+    const storedWithoutUpdatedAt = { ...storedOpportunity.record };
+    const inMemoryWithoutUpdatedAt = { ...opportunity.record };
+    delete storedWithoutUpdatedAt.updatedAt;
+    delete inMemoryWithoutUpdatedAt.updatedAt;
+
+    expect(storedWithoutUpdatedAt).to.eql(inMemoryWithoutUpdatedAt);
+  });
+
+  it('finds all opportunities by siteId', async () => {
+    const { Opportunity } = dataAccess;
+
+    const opportunities = await Opportunity.allBySiteId(siteId);
+
+    expect(opportunities).to.be.an('array').with.length(10);
+  });
+
+  it('creates a new opportunity', async () => {
+    const { Opportunity } = dataAccess;
+    const data = {
+      siteId,
+      auditId: uuid(),
+      title: 'New Opportunity',
+      description: 'Description',
+      runbook: 'https://example.com',
+      type: 'broken-backlinks',
+      origin: 'AI',
+      status: 'NEW',
+      data: { brokenLinks: ['https://example.com'] },
+    };
+
+    const opportunity = await Opportunity.create(data);
+
+    expect(opportunity).to.be.an('object');
+
+    expect(uuidValidate(opportunity.getId())).to.be.true;
+    expect(isIsoDate(opportunity.getCreatedAt())).to.be.true;
+    expect(isIsoDate(opportunity.getUpdatedAt())).to.be.true;
+
+    delete opportunity.record.opportunityId;
+    delete opportunity.record.createdAt;
+    delete opportunity.record.updatedAt;
+    expect(opportunity.record).to.eql(data);
+  });
+
+  it('deletes an opportunity', async () => {
+    const { Opportunity } = dataAccess;
+
+    const opportunity = await Opportunity.findById(sampleData.opportunities[0].getId());
+
+    await opportunity.remove();
```

Comment:
Would removing an opportunity also remove associated suggestions at the data access layer or would it need to be implemented at controller level?

Reply:
the `base.model.js` contains the todo `// todo: remove dependents (child associations)`. i'd like to tackle this in a subsequent PR if you don't mind.